connectors/inetsoft-odata/src/main/java/inetsoft/uql/odata/ODataRuntime.java

```
@@ -389,7 +389,14 @@ static Document getMetaDataDocument(ODataDataSource ds) {
          }
 
          try(CloseableHttpResponse httpResponse = httpClient.execute(getRequest)) {
-            return Tool.parseXML(httpResponse.getEntity().getContent());
+            if(httpResponse.getStatusLine().getStatusCode() >= 200 &&
+               httpResponse.getStatusLine().getStatusCode() < 300)
+            {
+               return Tool.parseXML(httpResponse.getEntity().getContent());
+            }
+            else {
+               LOG.warn("could not get metadata for datasource: " + ds.getName());
```

It is better to throw an exception and record the relevant information of the failure, rather than just recording a warning:
for example:
String errorResponse = EntityUtils.toString(httpResponse.getEntity());
LOG.warn("Failed to get metadata for datasource: {}, HTTP status: {}, Response: {}",
        ds.getName(), statusCode, errorResponse);
throw new RuntimeException("HTTP request failed with status " + statusCode);